services-camel-app/src/main/java/edu/unc/lib/boxc/services/camel/images/ImageDerivativeProcessor.java

```diff
@@ -47,6 +48,21 @@ public static boolean allowedImageType(Exchange exchange) {
         return true;
     }
 
+    /**
+     * Run jp24u kdu_compress command and generate image derivatives
+     * @param exchange
+     */
+    public void runJp24u(Exchange exchange) {
+        Message in = exchange.getIn();
+        String imagePath = (String) in.getHeader(CdrFcrepoHeaders.CdrImagePath);
+        String tempPath = (String) in.getHeader(CdrFcrepoHeaders.CdrTempPath);
+        String mimetype = (String) in.getHeader(CdrFcrepoHeaders.CdrBinaryMimeType);
+
+        String[] command = new String[]{"jp24u", "kdu_compress", "-f", "\"" + imagePath + "\"",
```

Comment:
I hadn't considered that we might still be able to interact with it like a CLI command. Calling it this way causes jp24u to execute within the same process? If so, it's probably simpler than directly calling the command classes

Reply:
According to chatgpt it should work so I figured I'd give it a shot. It looked a lot simpler than directly calling the command classes.